### PR TITLE
ui, tablemetadatacache: disable sorting on auto stats collection columns, expose batch size as a cluster setting

### PR DESCRIPTION
**Pull request by xinhaoz**
cluster-ui: disable sorting on auto stats columns

In the table overview page disable sorting on the
columns pertaining to auto stats collection. These
columns are not sortable.

Epic: CRDB-37558
Release note: None

------------------

tablemetadatacache: make the update job table batch size a CS

This commit makes the update job batch size adjustable
via a cluster setting.

Epic: none
Release note: None
@xinhaoz

### CODE REVIEW

File: pkg/sql/tablemetadatacache/cluster_settings.go [hunk 1]
```diff
@@ -34,3 +39,11 @@ var DataValidDurationSetting = settings.RegisterDurationSetting(
 		return nil
 	}),
 	settings.WithPublic)
+
+var updateJobBatchSizeSetting = settings.RegisterIntSetting(
+	settings.ApplicationLevel,
+	"obs.tablemetadata.update_job_batch_size",
+	"the number of tables to update in a single batch",
```
Comment (kyle-a-wong) [inline pkg/sql/tablemetadatacache/cluster_settings.go:46]: nit: maybe include "of the table metadata update job" or something

**Review by kyle-a-wong — APPROVED** [review]
Just a small nit about the cs description, otherwise lgtm

File: pkg/sql/tablemetadatacache/cluster_settings.go [hunk 2]
```diff
@@ -34,3 +39,11 @@ var DataValidDurationSetting = settings.RegisterDurationSetting(
 		return nil
 	}),
 	settings.WithPublic)
+
+var updateJobBatchSizeSetting = settings.RegisterIntSetting(
```
Comment (kyle-a-wong) [inline pkg/sql/tablemetadatacache/cluster_settings.go:43]: Are we intentionally not making this public?

Comment (xinhaoz) [reply pkg/sql/tablemetadatacache/cluster_settings.go:43]: Hmm good question let's ask Kevin if we want to expose this. I'll leave it as private for now.

**Comment by xinhaoz** [comment]
TFTR!
bors r+

**Pull request merged**